performance/plugins/performanceTestItem.ts

```diff
@@ -26,71 +34,162 @@ export class performanceTestItem implements TestItem {
+        nonce.iaddn(1);
+      });
+    });
+
+    ipc.server.start();
+    this.ipc = ipc;
 
     return true;
   }
+
   async act(testParams: TestParams): Promise<boolean> {
-    testLog.getLog().info("Act" + testParams);
+    const mga = await getMangata(testParams.nodes[0]!);
+    const api = await mga.getApi();
+    this.executed = trackExecutedExtrinsics(api, testParams.duration);
+    this.enqueued = trackEnqueuedExtrinsics(api, testParams.duration);
+    this.pending = trackPendingExtrinsics(api, testParams.duration);
     return true;
   }
-  async expect(nodes: string[]): Promise<boolean> {
-    //wait for not Txs pending + 10 blocks.
-    const mga = await getMangata(nodes[0]!);
-    let pendingExtr: any[] = [];
-    do {
-      const api = await mga.getApi();
-      pendingExtr = await api.rpc.author.pendingExtrinsics();
-    } while (pendingExtr.length > 0);
-    console.info(`Done waiting Txs!`);
-    console.info(`Stopping listeners....`);
-    this.listeners.forEach(async (p) => (await p).cancel());
-    console.info(`...Stopped`);
+
+  async expect(testParams: TestParams): Promise<boolean> {
+    let [executed, enqueued, pending] = await Promise.all([
+      this.executed,
+      this.enqueued,
+      this.pending,
+    ]);
+
+    writeToFile("executed.txt", executed);
+    writeToFile("enqueued.txt", enqueued);
+    writeToFile("pending.txt", pending);
+    generateHtmlReport("report.html", enqueued, executed, pending);
+
+    // ignore first few blocks
+    executed = executed.slice(5);
+    enqueued = enqueued.slice(5);
+    pending = pending.slice(5);
+
+    console.info(executed);
+    console.info(enqueued);
+    console.info(pending);
+
+    const execution_throughput = quantile(
+      executed.map(([_, val]) => val),
+      0.1
+    );
+    const enqueue_throughput = quantile(
+      enqueued.map(([_, val]) => val),
+      0.1
+    );
+    const pending_throughput = quantile(
+      pending.map(([_, val]) => val),
+      0.1
+    );
+
+    console.info(
+      `execution_thruput : 90% of measurements was above ${execution_throughput}`
+    );
+    console.info(
+      `enqueue_thruput   : 90% of measurements was above ${enqueue_throughput}`
+    );
+    console.info(
+      `pending_thruput   : 90% of measurements was above ${pending_throughput}`
+    );
+
+    if (execution_throughput < testParams.throughput) {
+      console.info(
+        `execution throughput was to low: ${execution_throughput} <= ${testParams.throughput}`
+      );
+      return false;
+    }
+
+    if (enqueue_throughput < testParams.throughput) {
+      console.info(
+        `enqueued throughput was to low: ${execution_throughput} <= ${testParams.throughput}`
+      );
+      return false;
+    }
+
+    if (pending_throughput < testParams.throughput * 0.75) {
+      console.info(
+        `pending throughput was too small, consider adding more threads`
+      );
+      return false;
+    }
```

Comment:
assert expected throughput

Reply:
I dont fully inderstand this :) 
+ Why are we comparing the same throughput for different  measurements?  :thinking:

Reply:
executed are just enqueued delayed by one block, but it can happen that for some reason some txs will stay in queue a bit longer (unexpected but possible) but at the end of a day we expect the same throughput as in ideal scenario these should be equal

Reply:
when it comes to pending, for performance test to make sens nodes should be under constant load - there should be some txs ready to fetch from the mempool all the time, 3/4 of expected load passed as parameter seems like good enough default. 

so assuming that node can execute N txs on every block on avarage
there should be at least 2 N txs available all the time
and as after new block is published (2N - 1N = N) there should be still N txs in queue waiting, but lets lower it down a bit

Reply:
when i think of it they could be renamed:
- pending -> throughput
- throughput -> expectedThroughput

would that work for you?